android/src/main/java/com/onegini/mobile/sdk/reactnative/handlers/pins/CreatePinRequestHandler.kt

```diff
@@ -6,61 +6,33 @@ import com.onegini.mobile.sdk.reactnative.Constants.PinFlow
 import com.onegini.mobile.sdk.android.model.entity.UserProfile
 import com.onegini.mobile.sdk.android.handlers.request.callback.OneginiPinCallback
 import com.onegini.mobile.sdk.android.handlers.error.OneginiPinValidationError
-import com.onegini.mobile.sdk.reactnative.Constants
-import com.onegini.mobile.sdk.reactnative.Constants.PIN_NOTIFICATION_CLOSE_VIEW
 
 class CreatePinRequestHandler : OneginiCreatePinRequestHandler {
     private var pinCallback: OneginiPinCallback? = null
-
-
-    private var pinNotificationHandler: PinNotificationObserver? = null
-    private var lastPinFlow: PinFlow = PinFlow.Create
-
-    fun setPinNotificationObserver(pinNotificationHandler: PinNotificationObserver?) {
-        this.pinNotificationHandler = pinNotificationHandler
-    }
+    private var eventEmitter = CreatePinEventEmitter()
 
     override fun startPinCreation(
         userProfile: UserProfile,
         oneginiPinCallback: OneginiPinCallback,
         pinLength: Int
     ) {
         pinCallback = oneginiPinCallback
-        notifyOnOpen(userProfile.profileId, pinLength)
+        eventEmitter.onPinOpen(userProfile.profileId, pinLength)
     }
 
     override fun onNextPinCreationAttempt(oneginiPinValidationError: OneginiPinValidationError) {
-            handlePinValidationError(oneginiPinValidationError)
+        eventEmitter.onError(oneginiPinValidationError.errorType, oneginiPinValidationError.message ?: "")
```

Comment:
I think you are right, and the same can be said about the authentication one. Let's refactor both at the same time as they are quite similar and leave this for now as it would require some extra RN changes aswell. 
I'll add a comment to the story for it.